app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js

```diff
@@ -120,35 +205,18 @@ async function switchEthereumChainHandler(
   requestData.fromNetworkConfiguration = getProviderConfig();
 
   if (requestData.toNetworkConfiguration) {
-    const currentChainId = getCurrentChainId();
-
     // we might want to change all this so that it displays the network you are switching from -> to (in a way that is domain - specific)
-
     const networkClientId = findNetworkClientIdByChainId(_chainId);
 
-    if (currentChainId === _chainId) {
-      if (hasPermissions(req.origin)) {
-        setNetworkClientIdForDomain(req.origin, networkClientId);
-      }
-      res.result = null;
-      return end();
-    }
-
     try {
       const approvedRequestData = await requestUserApproval({
         origin,
         type: ApprovalType.SwitchEthereumChain,
         requestData,
       });
-      if (
-        Object.values(BUILT_IN_INFURA_NETWORKS)
-          .map(({ chainId: id }) => id)
-          .includes(_chainId)
-      ) {
-        await setProviderType(approvedRequestData.type);
-      } else {
-        await setActiveNetwork(approvedRequestData.id);
-      }
```

Comment:
`setProviderType` is deprecated and `setActiveNetwork` can now handle both cases here